### ISSUE TITLE
Variables with leading underscores and/or containing forward slashes are invalid for SPSS

### ISSUE THREAD

**Issue by joshuaberetta** [kobotoolbox/formpack#246]
Need to handle the cases of:
- [ ] `_foo`
- [ ] `foo__bar`
- [ ] `foo/bar`
- [ ] ?

Ensure that:

> All future changes in the export file should correspond to the SPSS Label export

Forum discussions:
https://community.kobotoolbox.org/t/have-kpi-new-exports-to-excel-format-dates-and-numbers-as-such/2026/7
https://community.kobotoolbox.org/t/problems-converting-data-from-csv-to-spss/18695/4
https://community.kobotoolbox.org/t/kobo-outputs-numbers-are-stored-as-text/371/11
https://community.kobotoolbox.org/t/when-you-create-question-with-type-multiple-selection-not-support-for-spss-syntax-file/13122/8

**Comment by joshuaberetta** [comment]
moved to: https://www.notion.so/kobotoolbox/Improve-SPSS-export-remove-invalid-content-d667b6fc99974255b989286cfee48579

**Comment by jnm** [comment]
> All future changes in the export file should correspond to the SPSS Label export

I don't think we would ever do this for the generic XLSX or CSV exports: storing some kind of parallel form schema or weird "yes, we've made at least one SPSS labels export" state seems like a mess. Maybe we could create a set of rules for massaging variable (question) names for SPSS, and then apply those equally in the SPSS labels exports as well as in a new "XLSX for SPSS" export type :thinking:

The good news is that, thanks to your work, people can already exclude the leading underscore fields from their exports. The use of `_` as a delimiter for select-multiple questions looks a mistake of mine that could be easy to fix (one-liner plus unit test changes): https://github.com/kobotoolbox/formpack/blame/43daf1d1bd9ffecaeb0e66f8ffb5a957a6e9b9e2/src/formpack/utils/spss.py#L9. The big question, though, is what the separator should be. Our choices are limited:
> Variable names can be up to 64 bytes long, and the first character must be a letter or one of the characters @, #, or $. Subsequent characters can be any combination of letters, numbers, nonpunctuation characters, and a period (.).

https://github.com/kobotoolbox/kpi/issues/1607#issuecomment-415907468
